### PR TITLE
Small fix

### PR DESCRIPTION
**Pull request by sakehl**
Some small fixes I needed:
* Increases stack size to 128m: I was running into stack overflow errors again. But if you rather have dev not have such a big stack size, I'll keep these changes local.
* Added almost all math.h functions: All trigonometric are imported as abstract functions, with some properties attached. Mainly making sure code that uses these does not immediately break down.
* I tried to consistently remove `CPrimitiveType` from some places (ran into bugs otherwise not verifying, like inlined float functions which did not work)
* I want to use predicates in triggers, these changes allow this (if it sees a predicateApply in triggers, it does not use the fractional permission, but does use the predicate).
* Made some types lazy, fixing #1195 : Although in that issue there are still relatively slow passes for simplify we might want to look at before closing it completely.

### CODE REVIEW

**Comment by pieter-bos** [comment]
If you have a reproducer for an example needing so much stack memory I'd appreciate it :)

**Comment by sakehl** [comment]
So I have this file:

After parsing takes 10minutes, it gets to the verification. On which it stack overflows eventually after approx 5 minutes.

[SolveDirectionHalide-CB-P.zip](https://github.com/utwente-fmt/vercors/files/15260970/SolveDirectionHalide-CB-P.zip)

File: src/viper/viper/api/transform/ColToSilver.scala [hunk 1]
```diff
@@ -39,6 +39,8 @@ case class ColToSilver(program: col.Program[_]) {
   val currentMapGet: ScopedStack[col.MapGet[_]] = ScopedStack()
   val currentDividingExpr: ScopedStack[col.DividingExpr[_]] = ScopedStack()
 
+  var inTriggers = false
```
Comment (pieter-bos) [inline src/viper/viper/api/transform/ColToSilver.scala]: please use a `ScopedStack` for this, I plan to enable parallelism within rewriters at some point.

**Comment by pieter-bos** [comment]
I've added a couple flags to diagnose potential causes of parsing slowness [here](https://github.com/utwente-fmt/vercors/wiki/Parser-Analysis)

**Comment by sakehl** [comment]
> I've added a couple flags to diagnose potential causes of parsing slowness [here](https://github.com/utwente-fmt/vercors/wiki/Parser-Analysis)


> So I have this file:
> 
> After parsing takes 10minutes, it gets to the verification. On which it stack overflows eventually after approx 5 minutes.
> 
> [SolveDirectionHalide-CB-P.zip](https://github.com/utwente-fmt/vercors/files/15260970/SolveDirectionHalide-CB-P.zip)


Ah, excuse me, my terminology was off. I did not mean "parsing", I meant that the complete process of parsing + transformations takes about 10 minutes. Especially frtom the "adtNull" transformation and onward it takes a lot of time.

The actually parsing is like a few seconds.

**Comment by pieter-bos** [comment]
No worries, parsing performance is a recurring problem anyway. Seconds is already 5-6 orders of magnitude slower than it should be, really.

**Comment by sakehl** [comment]
ok should be ok now.

**Pull request merged**